test/roby_app/test_plugin.rb

```diff
@@ -313,6 +312,43 @@ def assert_upload_succeeds(client)
                     end
                 end
             end
+
+            describe "Log Rotation" do
+                before do
+                    Syskit.conf.log_rotation_period = nil
+                    @app = Roby::Application.new(plan: plan)
+
+                    task_m = Syskit::TaskContext.new_submodel
+                    task_m.provides Syskit::LoggerService
+                    @task = syskit_stub_deploy_configure_and_start(task_m)
+
+                    @logfile = []
+                    2.times { @logfile << create_test_file(make_tmpdir) }
+
+                    @rotated_logs = {}
+                    flexmock(@task).should_receive(:rotate_log)
+                                  .explicitly.and_return(@logfile)
+                    @app.rotate_logs(@rotated_logs)
+                end
+
+                it "rotates log" do
+                    assert_equal @rotated_logs[@task.execution_agent.arguments[:on]], @logfile
+                end
+
+                it "verifies single process error in remote process server" do
+                    @rotated_logs.each do |key, value|
+                        assert(value.length() > 1, "identified single process per process server")
```

Comment:
Don't simply check the count... Check the content !

Reply:
does the "rotates log" test case already satisfy the content checking?